store/sqlstore/status_store.go

```diff
@@ -39,20 +39,19 @@ func (s SqlStatusStore) CreateIndexesIfNotExists() {
 	s.CreateIndexIfNotExists("idx_status_status", "Status", "Status")
 }
 
-func (s SqlStatusStore) SaveOrUpdate(status *model.Status) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if err := s.GetReplica().SelectOne(&model.Status{}, "SELECT * FROM Status WHERE UserId = :UserId", map[string]interface{}{"UserId": status.UserId}); err == nil {
-			if _, err := s.GetMaster().Update(status); err != nil {
-				result.Err = model.NewAppError("SqlStatusStore.SaveOrUpdate", "store.sql_status.update.app_error", nil, err.Error(), http.StatusInternalServerError)
-			}
-		} else {
-			if err := s.GetMaster().Insert(status); err != nil {
-				if !(strings.Contains(err.Error(), "for key 'PRIMARY'") && strings.Contains(err.Error(), "Duplicate entry")) {
-					result.Err = model.NewAppError("SqlStatusStore.SaveOrUpdate", "store.sql_status.save.app_error", nil, err.Error(), http.StatusInternalServerError)
-				}
+func (s SqlStatusStore) SaveOrUpdate(status *model.Status) (err *model.AppError) {
```

Comment:
We prefer the explicit return instead of the variable return approach. Do you mind to change it?